server/bleep/src/semantic/chunk.rs

```diff
@@ -172,6 +177,13 @@ fn add_token_range<'s>(
     if trimmed_end_byte <= start_byte {
         return;
     }
+    assert!(
```

Comment:
Rather than `assert` here, we could `warn`, and truncate the number of tokens produced by the tokenizer. In `semantic.rs` we can put something like:

```
 let mut tokenizer =
            tokenizers::Tokenizer::from_file(model_dir.join("tokenizer.json")).unwrap();
        tokenizer.with_padding(Some(PaddingParams {
            strategy: PaddingStrategy::BatchLongest,
            direction: PaddingDirection::Right,
            pad_id: 0,
            pad_type_id: 0,
            pad_token: "[PAD]".into(),
            pad_to_multiple_of: None,
        }));
```
when initialising the tokenizer.

Reply:
The problem with that is that, again, we don't have the actual number of tokens, and our calculation is merely a guess. In this case, the assert was there to ensure that we never under any circumstance get an overlong chunk. I'd be fine with using a `debug_assert!(..)`  instead.

Reply:
I suppose the problem with truncating at the tokeniser level is that we'd have to toggle truncation on-and-off depending on whether we are chunking or embedding. A better solution would be to not re-use tokens calculated when chunking, passing them to the embedder. We can address this in a future issue.